daemon/api_debug.go

```diff
@@ -128,7 +135,69 @@ func getChangeTimings(st *state.State, changeID string) Response {
 			UndoingTimings: undoingTimingsByTask[t.ID()],
 		}
 	}
-	return SyncResponse(m, nil)
+	return m, nil
+}
+
+func collectEnsureTimings(st *state.State, ensureTag string, allEnsures bool) ([]*debugTimings, error) {
+	ensures, err := timings.Get(st, -1, func(tags map[string]string) bool {
```

Comment:
I looked at `timings.Get`, wondering about the case when `allEnsures == false`, when we grab/unmarshal/filter all the entries, but only push the last one out. Given that there will always be quite a bit of timings data, must of the work is being discarded. OTOH, maybe it's not a problem after all.

Reply:
I don't think performance it's a concern as this is a debug interface, also I'm not sure this can be avoided given that we need to look into tags. FWTW we currently keep up to 100 "root-level" timings (but yes, they can have a few nested timings each).